src/js/App.ts

```diff
@@ -0,0 +1,144 @@
+import Component from "@/libs/component";
+import Header from "@/components/Header";
+import SearchModal from "@/components/SearchModal";
+import ClassRoom from "@/components/ClassRoom";
+import videoDB from "@/libs/videoDB";
+import parseVideoData from "@/utils/parseVideoData";
+import popUpSnackBar from "@/utils/popUpSnackBar";
+import { $ } from "@/utils/dom";
+import { AppState, Navigations, Item } from "@/types/index";
+import {
+  APP_SELECTORS,
+  FILTER_ID,
+  SEARCH_BUTTON_ID,
+  TOAST_MESSAGES,
+} from "@/constants/index";
+
+class App extends Component<AppState> {
+  $headerComponent: Component | null = null;
+  $searchModalComponent: Component | null = null;
+  $classRoomComponent: Component | null = null;
+
+  constructor($root: HTMLElement) {
+    super();
+    this.$root = $root;
```

Comment:
```suggestion
  constructor(
    public readonly $root: HTMLElement
  ) {
    super();
```
이렇게도 표현할 수 있답니다! 이러면 생성자로 받아온 매개변수가 자동으로 멤버변수에 할당됩니다.